### PR TITLE
#3017 Deprecate the Block Content Permissions module and no longer install by default

### LINKED ISSUE #3017: Deprecate the Block Content Permissions module and no longer install by default
## Motivation
The maintainers of the [Block Content Permissions](https://www.drupal.org/project/block_content_permissions) module have stated that it is no longer needed as of Drupal 10.1.x-dev.

## Proposed resolution
The module maintainers recommend migrating and setting up new permissions (as described on the [module project page](https://www.drupal.org/project/block_content_permissions)), disabling the module, testing that roles have access as expected, and then uninstalling the module.

Our steps for this process will be:

1. In this issue, we can deprecate the module and prevent it from being installed by default with Quickstart.
2. In [issue #3014](https://github.com/az-digital/az_quickstart/issues/3014), we can make the necessary changes to user role permissions.
3. In a future issue, we can disable and uninstall the module from Quickstart sites.

## Additional context
- Quickstart issue: [Revisit block administration permissions for content admin/editor roles](https://github.com/az-digital/az_quickstart/issues/3014)
- Drupal core issue: [Add more granular block content permissions](https://www.drupal.org/project/drupal/issues/1975064)

### PR DESCRIPTION
**Pull request by bberndt-uaz**
## Description

Deprecates the [Block Content Permissions module](https://www.drupal.org/project/block_content_permissions) and changes Quickstart to no longer install it by default since the module is no longer needed as of Drupal 10.1 (Quickstart 2.8).

## Related issues

#3017 

## How to test

Link to Probo review site: https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-3020.probo.build

1. Confirm that the Block Content Permissions module was not installed by default.
2. After installing the module, confirm that the Status Report page shows a "Quickstart deprecated modules enabled" warning for the module.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [x] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: config/install/user.role.az_content_admin.yml [hunk 1]
```diff
@@ -33,7 +33,6 @@ permissions:
   - 'bypass node access'
   - 'create content translations'
   - 'create new books'
-  - 'create az_custom_menu_block block content'
```
Comment (joeparsons) [inline config/install/user.role.az_content_admin.yml:36]: FWIW, the new `create xxxx block content` permissions provided by core as of Drupal 10.1 has the same machine names so I'll probably be adding this one back in my PR for #3014.

File: config/install/user.role.az_content_editor.yml [hunk 2]
```diff
@@ -41,13 +41,11 @@ permissions:
   - 'create az_course content'
   - 'create az_document media'
   - 'create az_event content'
-  - 'create az_flexible_block block content'
```
Comment (joeparsons) [inline config/install/user.role.az_content_editor.yml:44]: FWIW, the new create xxxx block content permissions provided by core as of Drupal 10.1 has the same machine names so I'll probably be adding this one back in my PR for https://github.com/az-digital/az_quickstart/issues/3014.

File: config/install/user.role.az_content_editor.yml [hunk 3]
```diff
@@ -41,13 +41,11 @@ permissions:
...
   - 'create az_flexible_page content'
   - 'create az_image media'
   - 'create az_news content'
   - 'create az_person content'
   - 'create az_publication content'
-  - 'create az_quick_links block content'
```
Comment (joeparsons) [inline config/install/user.role.az_content_editor.yml:50]: FWIW, the new create xxxx block content permissions provided by core as of Drupal 10.1 has the same machine names so I may be adding this one back in my PR for https://github.com/az-digital/az_quickstart/issues/3014.

**Comment by joeparsons** [comment]
Discussion topic: Do we need a DB update to programmatically remove the permissions provided by this module from our roles?

If we don't do so the roles probably risk getting deleted when the module is uninstalled on an existing site (?)

**Review by trackleft — APPROVED** [review]

**Comment by bberndt-uaz** [comment]
Reassigning to @joeparsons since I will be out next week. @tadean agreed that a DB update might be a good idea.

**Comment by joeparsons** [comment]
Going to combine this with #3027.

**Pull request merged**